app/main.py

```diff
@@ -1,8 +1,38 @@
 class Person:
-    # write your code here
-    pass
+    people = {}
+
+    def __init__(self, name : str, age : int) -> None:
+        self.name = name
+        self.age = age
+        self.people[name] = Person
 
 
 def create_person_list(people: list) -> list:
```

Comment:
To streamline the algorithm, consider simplifying it. You can achieve the desired outcome using just two loops. Begin by creating instances for people within the first loop, leaving the wife and husband fields unfilled initially. Then, in the second loop, assign these fields accordingly.